upload/src/addons/TickTackk/DeveloperTools/Cli/Command/EntityFromTable.php

```diff
@@ -0,0 +1,426 @@
+            }
+
+            if ($allowedValues)
+            {
+                $fieldData['allowedValues'] = $allowedValues;
+            }
+            if (isset($fieldData['Null']) && $fieldData['Null'] !== 'NO')
+            {
+                $fieldData['nullable'] = var_export(true, true);
+            }
+            if (isset($colDefinition['Default']) && ($colDefinition['Default'] !== null || !empty($fieldData['nullable'])))
+            {
+                $default = $colDefinition['Default'];
+                switch($type)
+                {
+                    case 'self::INT':
+                    case 'self::UINT':
+                        if (\is_scalar($default))
+                        {
+                            $default = intval($default);
+                        }
+                        break;
+                    case 'self::FLOAT':
+                        if (\is_scalar($default))
+                        {
+                            $default = strval(floatval($default)) + 0;
+                        }
+                        break;
+                    case 'self::BOOL':
+                        $default = intval($default) ? true : false;
+                        break;
+                }
+
+                $fieldData['default'] = var_export($default, true);
+            }
+
+            if (!\array_key_exists('default', $fieldData) && empty($fieldData['nullable']))
+            {
+                $fieldData['required'] = var_export(true, true);
+            }
+
+            $definition = [];
+            foreach ($fieldData as $key => $value)
+            {
+                $definition[] = var_export($key, true) . " => " . $value;
+            }
+            $definition = implode($definition, ", ");
+
+            $columns .= "            " . var_export($colDefinition['Field'], true) . " => [" . $definition . "],\n";
+        }
+
+
+        $template = <<<TEMPLATE
+<?php
+
+namespace {$namespace}\Entity;
+
+use XF\Mvc\Entity\Entity;
+use XF\Mvc\Entity\Structure;
+
+class {$name} extends Entity
+{
+    /**
+     * @param Structure \$structure
+     * @return Structure
+     */
+    public static function getStructure(Structure \$structure)
+    {
+        \$structure->table = '{$table}';
+        \$structure->shortName = '{$namespace}:{$name}';
+        \$structure->primaryKey = {$primaryKey};
+        \$structure->columns = [
+{$columns}        ];
+
+        return \$structure;
+    }
+}
+TEMPLATE;
+        $entityName = "{$namespace}:{$name}";
+        $entityName = str_replace('/', '\\', $entityName);
+
+        $output->writeln("Writing entity for {$entityName}");
+echo $template."\n\n";
+        File::writeFile($filename, $template, false);
+
+        $this->runSubAction($output, [
+            'xf-dev:entity-class-properties',
+            $entityName
+        ]);
+
+        $this->runSubAction($output, [
+            'xdt-addon:generate-schema-entity',
```

Comment:
Should `'tdt-addon:generate-schema-entity'` 😝

Reply:
Fixed!